source/Octopus.Client.Tests/Exceptions/OctopusExceptionFactoryFixture.cs

```diff
@@ -0,0 +1,321 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using Octopus.Client.Exceptions;
+
+namespace Octopus.Client.Tests.Exceptions
+{
+    public class OctopusExceptionFactoryFixture
+    {
+        [Test]
+        [TestCase(HttpStatusCode.BadRequest)]
+        [TestCase(HttpStatusCode.Conflict)]
+        public async Task Http400or409Response_WithNoPayLoad_ShouldCreateOctopusValidationException(
+            HttpStatusCode statusCode)
+        {
+            var httpResponseMessage = new HttpResponseMessage(statusCode);
+            var createdException = await OctopusExceptionFactory.CreateException(httpResponseMessage);
+
+            using (new AssertionScope())
+            {
+                createdException.Should().BeOfType<OctopusValidationException>();
+                createdException.HttpStatusCode.Should().Be((int)statusCode);
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(CreatedOctopusValidationExceptionTestData))]
+        public async Task CreatedOctopusValidationException_ShouldContainPayloadInformation(string payload, OctopusValidationException expectedResult)
+        {
+            var httpResponseMessage = new HttpResponseMessage(HttpStatusCode.BadRequest);
+
+            if (!string.IsNullOrWhiteSpace(payload))
+            {
+                httpResponseMessage.Content = new StringContent(payload);
+            }
+
+            var createdException = await OctopusExceptionFactory.CreateException(httpResponseMessage) as OctopusValidationException;
+            createdException.Should().NotBeNull();
+
+            using (new AssertionScope())
+            {
+                // ReSharper disable once PossibleNullReferenceException
+                createdException.ErrorMessage.Should().Be(expectedResult.ErrorMessage);
+                createdException.Errors.Should().BeEquivalentTo(expectedResult.Errors);
+                createdException.HelpText.Should().Be(expectedResult.HelpText);
+            }
+        }
+
+        public static IEnumerable<TestCaseData> CreatedOctopusValidationExceptionTestData()
+        {
+            const int httpStatusCode = (int)HttpStatusCode.BadRequest;
+            const string errorMessageValue = "Error Message";
+            const string helpTextValue = "Help Text";
+            var errorsValue = new[] { "Additional Errors" };
+            var jObject = new JObject
+            {
+                { "ErrorMessage", errorMessageValue },
+                { "Errors", JArray.FromObject(errorsValue) },
+                { "HelpText", helpTextValue },
+                { "Random", "not relevant"}
+            };
+
+            yield return new TestCaseData(jObject.ToString(), new OctopusValidationException(httpStatusCode, errorMessageValue, errorsValue) { HelpText = helpTextValue });
```

Comment:
It's a clever way to generate the test data, but I think it'd be better to be actually _see_ the JSON. I can guess pretty well what the result of `jObject.ToString()` would be, but I'm not exactly sure. Might just be my personal preference, so I won't hold the PR up with it.

Reply:
I prefer this because injecting const variable on JSON string can get a bit messy. It's an eye sore as well with the amount of quotes it needs.